doozerlib/exectools.py

```diff
@@ -118,9 +122,45 @@ def cmd_gather(cmd, set_env=None):
             cmd_info, exc, cmd_list[0]
         ))
         return exc.errno, "", "See previous error description."
-    out, err = proc.communicate()
-    rc = proc.returncode
+
+    if not realtime:
+        out, err = proc.communicate()
+        rc = proc.returncode
+    else:
+        out = ""
+        err = ""
+
+        # Many thanks to http://eyalarubas.com/python-subproc-nonblock.html
```

Comment:
I <3 it when people give proper credit to external resources, like you did here. :hugs:

Reply:
:D If I didn't I wouldn't remember where I got it from.